compiler/noirc_errors/src/position.rs

```diff
@@ -51,27 +81,49 @@ impl<T> std::borrow::Borrow<T> for Spanned<T> {
     }
 }
 
+#[derive(
+    Copy, Clone, Serialize, Eq, PartialEq, Ord, PartialOrd, Debug, Deserialize, Hash, Default,
+)]
+pub struct SrcId(usize);
+
+impl From<usize> for SrcId {
+    fn from(value: usize) -> Self {
+        Self(value)
+    }
+}
+
+impl From<SrcId> for usize {
+    fn from(val: SrcId) -> Self {
+        val.0
+    }
+}
+
 #[derive(
     PartialEq, PartialOrd, Eq, Ord, Hash, Debug, Copy, Clone, Default, Deserialize, Serialize,
 )]
-pub struct Span(ByteSpan);
+pub struct Span(ByteSpan, SrcId);
 
 impl Span {
-    pub fn inclusive(start: u32, end: u32) -> Span {
-        Span(ByteSpan::from(start..end + 1))
+    pub fn inclusive(start: Position, end: Position) -> Span {
+        Span(ByteSpan::from(start.0..end.0 + 1), start.1)
+    }
+
+    pub fn inclusive_within(start: u32, end: u32, src_id: impl Into<usize>) -> Span {
+        Span(ByteSpan::from(start..end + 1), src_id.into().into())
     }
 
-    pub fn single_char(start: u32) -> Span {
-        Span::inclusive(start, start)
+    pub fn single_char(start: u32, src_id: impl Into<usize>) -> Span {
+        let start_position = Position::new(start, src_id.into().into());
+        Span::inclusive(start_position, start_position)
```

Comment:
```suggestion
    pub fn single_char(start: u32, src_id: SrcId) -> Span {
        let start_position = Position::new(start, src_id);
        Span::inclusive(start_position, start_position)
```
`.into().into()` is definitely too far for me. If we need a `SrcId`, lets have the function take one directly

Reply:
applied